### DISCUSSION TITLE
How to View Old Logs?

### DISCUSSION THREAD

**Discussion by kxzo** [vercel/vercel#9640] [Help] [answered]
I guess the "Runtime Logs" are new, but they aren't well thought. Like, I have a lot of "breakpoints", where I simply log data with console.log() but now I can't even see the log content, just the status code, heading, etc.

How can I again view the "old" logs? (with my real console.logs)

Now so 😥:
![image](https://user-images.githubusercontent.com/66639485/224134680-ee416c60-6716-4b86-9526-4182ab618d2f.png)

**Comment by lostra01** [comment]
Same issue here. With an update we ran into some issues, none of our old console.log or errors work anymore, so there's absolutely ZERO way to debug it, as it's not happening in local, only on Vercel.

**Comment by vvo** [comment]
Hey there @einKnuffy and @lostra01, could you send more details about this issue like:
- are you using a specific logging library?
- @einKnuffy, I see "info - Loaded env from /var/task/.env", this definitely looks like custom logging so let's see why the next lines are not showing up
- is there a way for you to reproduce the issue in a public git repository that we can have a look at? Thanks!

**Reply by lostra01** [reply]
Hi @vvo ! There's no custom logging library at all. Our application runs on NodeJS. In your previous log system, when we had any kind of console.log, console.warn or console.error, it showed up on Vercel logs, so we knew exactly what the issue was if there was any. 

Now the code fails at some point (SSR + Vercel Adapter on Astro), but with the new log system you have in place, we can only see the call headers, and that it has failed. Our previously logged console entries are nowhere to be found. 

Adding a complete custom logging library that sends the log data to some 3rd party service or pushes it into a DB every step of the way instead of simply using the console to print any information is incredibly counter-intuitive.

**Reply by lostra01** [reply]
Attaching the image of how the logs you have in place look now. In the background there's about 10 log entries that should be printed before it reaches any point that can cause the site to break. As you can see none of that show up anymore. With the old log system, it did.. 

<img width="1390" alt="Screenshot 2023-03-14 at 14 32 57" src="https://user-images.githubusercontent.com/9054858/225017226-f21f2a47-b9e8-4aae-abcd-406b574416b2.png">

**Reply by kxzo** [reply]
@vvo Yeah, as I already told and as @lostra01 told, this is neither a bug report nor any code issue, it is just about the fact that debugging deployed and running website applications is impossible with new logging system. And the question again was, firstly how can return to the previous logging system with having detailed console outputs, etc...

**Comment by wildky** [comment]
I thought maybe it was just bad UX and I couldn't find the button to view log messages (ex: `console.log()`) within a Serverless Function execution. @vvo Have log messages seriously been removed from the new Vercel Logs tool?? 

This is a critical feature for our organization when it comes to debugging, please bring it back. 😞

**Comment by vvo** [comment]
To be clear for anyone reading this issue: we did not remove console logs from the new Vercel Logs. If there are situations where the logs are not appearing, and you can create a reproduction repository + steps that we can install and try please do that and we will have a look.

Here's an example Next.js project I just deployed and got logs for: 
![CleanShot 2023-03-16 at 4 59 24](https://user-images.githubusercontent.com/123822/225679540-0dbfc9ee-44a6-408a-a6b2-17096dfad0e6.png)
<img width="956" alt="CleanShot 2023-03-16 at 5 00 33" src="https://user-images.githubusercontent.com/123822/225679583-463eec12-d8e0-4d55-a661-46e0dc24f7fb.png">

**Comment by vvo** [comment]
>  And the question again was, firstly how can return to the previous logging system with having detailed console outputs, etc...

There's no way at the moment to go back to the previous logging system. We will find a way together to resolve the issue of logs not showing up for you and see how that goes.

**Comment by vvo** [comment]
Finally, here's the live mode in action to follow logs coming in for a specific function: 

https://user-images.githubusercontent.com/123822/225680527-83f63a85-2dda-45a2-9f95-f1e5e2e972b9.mp4

**Comment by KeyToLove** [comment]
@vvo Hi there, just like the below screenshots, i declarant a large data and log it in `serverless functions`
![image](https://user-images.githubusercontent.com/54970106/225794611-9d01de46-7aaa-4a00-8716-ce57bb68e0ff.png)
But the log has be folded,How can I get the full log information? This is very important for troubleshooting.
![image](https://user-images.githubusercontent.com/54970106/225795304-29bb3656-d099-4ade-b22a-9ef50d1ba5e2.png)

**Reply by vvo** [reply]
@KeyToLove This is a separate concern from the current discussion. But to solve this, you can do:

```ts
console.log(JSON.stringify(details));
```

and you will get the full output.
Please note that Vercel logs are still limited to 4kb so you may still not get the full output.

**Comment by wildky** [comment, marked as answer]
@vvo Thanks for clarifying that log messages are still printed in the new Vercel Logs tool. 

Today when I check again, I see function logs now, as I expect to see them. To be fair, there is a chance I was just disoriented by the new UI yesterday, and at this point I don't have any issues, nor any way to reproduce the issue I was having yesterday. Thanks for the quick feedback!
